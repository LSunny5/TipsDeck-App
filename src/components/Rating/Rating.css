.rating {
    color: blue;
    padding: 0 1px;
  }