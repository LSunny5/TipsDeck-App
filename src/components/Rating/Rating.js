import React from 'react';
import './Rating.css';

//component that displays the stars according to the rating
export default function Rating(props) {
  const stars = [0,0,0,0,0]
    .map((_, i) => i < props.value
      ? <span key={i}>&#9733;</span>
      : <span key={i}>&#9734;</span>
    );
  return (
    <div className="rating">
      {stars}
      <p>{props.value} stars</p>
    </div>
  );
}