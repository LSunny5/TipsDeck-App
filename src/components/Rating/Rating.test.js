import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Rating from './Rating';

describe(`Rating component`, () => {
    it('renders the Rating by default', () => {
        const wrapper = shallow(<Rating />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
