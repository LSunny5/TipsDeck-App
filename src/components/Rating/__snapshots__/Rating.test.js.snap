// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component renders the Rating by default 1`] = `
<div
  className="rating"
>
  <span
    key="0"
  >
    ☆
  </span>
  <span
    key="1"
  >
    ☆
  </span>
  <span
    key="2"
  >
    ☆
  </span>
  <span
    key="3"
  >
    ☆
  </span>
  <span
    key="4"
  >
    ☆
  </span>
  <p>
     stars
  </p>
</div>
`;
